Design/Packaging/Future_packaging_strategy.md

```diff
@@ -0,0 +1,70 @@
+<!-- This document's diagrams are produced using the .plantuml documents in this file's directory and plantuml -->
+# Future packaging with conda
+## Current methodology
+At present we have bespoke methodologies for packaging on each different OS we support (Windows, MacOSX, Ubuntu, and CentOS/RHEL7).
+
+- On Windows, we create a packaged installer containing our dependencies and our compiled program as a .exe. We only link to some parts of System32 that provide Windows functionality required for Mantid, and the OpenGL shim to handle the graphics drivers. We install Mantid running the .exe installer by double clicking on the executable.
+
+- On MacOSX, we create a package containing our dependencies and our compiled program in a .dmg app bundle. Only low level packages that are expected to exist on MacOSX. We install Mantid .dmg by running the installer by double clicking on the bundle.
+
+- On Ubuntu, we create a package containing only our compiled program in a .deb format. We rely on the the currently available system packages to run Mantid. We install the Mantid .deb package using dpkg.
+
+- On CentOS/RHEL7, we create a package containing only our compiled program in a .rpm format. We rely on the currently available system packages, optional packages from `rhel-7-workstation-optional-rpms` and some packages from EPEL to run Mantid. We install the Mantid .rpm package using the rpm command.
+
+## Intended move forward
+
+All operating systems should have a fully built and deployable conda-build implementation, where a use can in theory run a simple command in conda and it will install Mantid with all it's runtime dependencies, which with some small tweaks we can then package on all Operating Systems assuming that the conda-build recipe has been created and ran successfully.
+
+The aim at this time is to create similar packages to what we have now on MacOSX and Windows, whilst shipping all dependencies in our Ubuntu and CentOS/RHEL7 which is a change. We will make a .exe for Windows, .dmg for MacOSX, .deb for Ubuntu, and a .rpm for CentOS/RHEL7 much like we do now on all operating systems.
+
+### Conda build
+
+We intend to distribute a conda package using the conda-build system, which will allow advanced users to install Mantid using conda. Users will be able to install this packaged version with, 
+```bash
+conda install -c mantid mantid-workbench
+```
+The user will then be able to interact with the package in a manner familiar to any users of other conda packages, e.g. Spyder, Numpy,
+```bash
+conda install -c mantid mantid-workbench // Install conda package
+python -c "import mantid"  // Import mantid
+MantidWorkbench // Launch workbench
```

Comment:
I'm certain that some people will disagree with pascal case for launching the program. Especially on linux.

Reply:
What would you suggest as an alternative? This was just an example I came up with at the time. Would `mantid-workbench` or `mantidworkbench` work better do you think?

Reply:
I think we can work this out later, right? We use PascalCase on Windows/MacOS but we can keep all lower case on linux.